backend/src/org/commcare/xml/EntryParser.java

```diff
@@ -97,10 +94,69 @@ public Entry parse() throws InvalidStructureException, IOException, XmlPullParse
             throw new InvalidStructureException("<entry> block must define display text details", parser);
         }
 
-        if (isEntry) {
+        if (FORM_ENTRY_TAG.equals(parserBlockTag)) {
             return new FormEntry(commandId, display, data, xFormNamespace, instances, stackOps, assertions);
-        } else {
+        } else if (VIEW_ENTRY_TAG.equals(parserBlockTag)) {
             return new ViewEntry(commandId, display, data, instances, stackOps, assertions);
+        } else if (SYNC_REQUEST_TAG.equals(parserBlockTag)) {
+            return new SyncEntry(commandId, display, data, instances, stackOps, assertions, post, queries);
+        }
+
+        throw new RuntimeException("Misconfigured entry parser");
```

Comment:
Nitpick, but this could be more useful by including what the given parserBlockTag was.